File: py_solutions_1-10/Euler_9.py
# Find the only pythagorean triplet that a + b + c = 1000 

from __future__ import print_function

import math
import timeit

try:
    range = xrange
except NameError:
    pass

start = timeit.default_timer()

def euler_9():
    for a in range(1, 3000):
        b = a + 1
        for b in range(b, 3000):
            c = sum([a, b, (math.sqrt(a**2 + b**2))])
            if c == 1000:
                return int(a * b * (math.sqrt(a**2 + b**2)))
            elif c > 1000:
                break

print('Answer: {}'.format(euler_9()))
stop = timeit.default_timer()
print('Time: {0:9.5f}'.format(stop - start))

File: Rust/Euler_9.rs
// Find the only pythagorean triplet that a + b + c = 1000 

use std::time::Instant;


fn main() {
    let timer = Instant::now();
    let mut answer = 0;

    for x in 1..1000 as u64 {
        for y in x+1..1000 as u64 {
            let tmp = (x*x + y*y) as f64;
            let zi = tmp.sqrt();
            let z = zi as u64;
            if z == zi && x + y + z == 1000 {
                println!("{} + {} + {}", x, y, z);
                answer = x * y * z;
                break;
            }
            if answer != 0 {
                break;
            }
        }
    }

    println!("Answer: {}", answer);
    println!("Time: {:?}", timer.elapsed());
}
